.github/workflows/node-test.yaml

```diff
@@ -85,10 +85,20 @@ jobs:
     steps:
```

Comment:
does this workflow have privileges such that hardening has any benefit at all? Most "test" workflows do not.

Reply:
I think it depends on the calling workflow permissions. In any case, as the linked article suggests, there are benefits to pinning besides the security aspects (same reason the ecosystem adopted lockfiles)

Reply:
There are also a ton of harms to pinning, which is why many projects, including express, explicitly ban pinning actions unless a workflow has dangerous privileges. It should _not_ be naively employed across a swath of projects.

Reply:
Does that mean that if this PR means, Express would have to pin this action to avoid the pinning? lol

Reply:
lol maybe. but srsly, i do think this is not a good thing to do for any non-dangerous workflows.